Polkadot Protocol > Architecture > Polkadot Chain > Overview (#115)
polkadot-developers/polkadot-docs: polkadot-protocol/architecture/polkadot-chain/overview.md

Reviewer: Some Polkadot design choices can be added from the following slides from PBA-book
https://polkadot-blockchain-academy.github.io/pba-book/polkadot/intro/page.html
https://polkadot-blockchain-academy.github.io/pba-book/polkadot/decisions/page.html

Reviewer 2: @0xLucca was this done? Can this be resolved?

Reviewer: All done

---
title: Overview
description: Explore Polkadot's core architecture, including its multi-chain vision, shared security, runtime upgrades, and the DOT token's governance and staking roles.
---

# Overview

## Introduction

> Polkadot is a next-generation blockchain protocol designed to support a multi-chain future by enabling secure communication and interoperability between different blockchains. Built as a Layer-0 protocol, Polkadot introduces innovations like application-specific Layer-1 chains (parachains), shared security through Nominated Proof-of-Stake (NPoS), and seamless cross-chain interactions via its native Cross-Consensus Messaging Format (XCM). 

Reviewer 2: ```suggestion
Polkadot is a next-generation blockchain protocol designed to support a multi-chain future by enabling secure communication and interoperability between different blockchains. Built as a Layer-0 protocol, Polkadot introduces innovations like application-specific Layer-1 chains (parachains), shared security through Nominated Proof of Stake (NPoS), and seamless cross-chain interactions via its native Cross-Consensus Messaging Format (XCM). 
```

Below we don't use hyphens, and I think majority of the rest of the docs don't. We should probably check and make any updates to ensure we're being consistent

- **Rapid L1 development** - the [Polkadot SDK](link to Polkadot SDK page in polkadot protocol section) allows fast, flexible creation and deployment of Layer-1 chains

- **Cultivating the next generation of Web3 developers** - Polkadot supports the growth of Web3 core developers through initiatives such as:

    - [Polkadot Blockchain Academy](https://polkadot.com/blockchain-academy){target=\_blank}
    - [Polkadot Alpha Program](https://polkadot.com/alpha-program){target=\_blank}
    - [EdX courses](https://www.edx.org/school/web3x){target=\_blank}
    - Rust and Substrate courses (coming soon)

### High-Level Architecture

Polkadot features a chain that serves as the central component of the system. This chain is depicted as a ring encircled by several parachains that are connected to it. 

According to Polkadot's design, any blockchain that can compile to WebAssembly (Wasm) and adheres to the Parachains Protocol becomes a parachain on the Polkadot network.

Here’s a high-level overview of the Polkadot protocol architecture:

![](/images/polkadot-protocol/architecture/polkadot-chain/overview/overview-1.webp)

> Parachains propose blocks to Polkadot validators, who check for availability and validity before finalizing them. With the chain providing security, collators—full nodes of parachains—can focus on their tasks without needing strong incentives.

Reviewer 2: ```suggestion
Parachains propose blocks to Polkadot validators, who check for availability and validity before finalizing them. With the relay chain providing security, collators—full nodes of parachains—can focus on their tasks without needing strong incentives.
```

Suggestion from Bader that was missed

> The [Cross-Consensus Messaging Format (XCM)](add link to XCM page){target=\_blank} allows parachains to exchange messages freely, leveraging the chain's security for trust-free communication.

Reviewer 2: ```suggestion
The [Cross-Consensus Messaging Format (XCM)](todo: add link to XCM page){target=\_blank} allows parachains to exchange messages freely, leveraging the chain's security for trust-free communication.
```

Reviewer 2: So it's easier to find

> In order to interact with chains that want to use their own finalization process (e.g., Bitcoin), Polkadot has [bridges](add link to bridges pages, Snowbridge, etc){target=\_blank} that offer two-way compatibility, meaning that transactions can be made between different parachains.

Reviewer 2: ```suggestion
In order to interact with chains that want to use their own finalization process (e.g., Bitcoin), Polkadot has [bridges](todo: add link to bridges pages, Snowbridge, etc){target=\_blank} that offer two-way compatibility, meaning that transactions can be made between different parachains.
```

### Polkadot's Additional Functionalities

> The Polkadot chain oversees crowdloans and auctions. Chain cores were leased through auctions for three-month periods, up to a maximum of two years. 

Reviewer 2: ```suggestion
The Polkadot relay chain oversees crowdloans and auctions. Chain cores were leased through auctions for three-month periods, up to a maximum of two years. 
```
Suggestion from Bader that was missed

Crowdloans enabled users to securely lend funds to teams for lease deposits in exchange for pre-sale tokens, which is the only way to access slots on Polkadot 1.0.

!!!note
    Auctions are deprecated in favor of [coretime](todo: update link).

Additionally, the chain handles [staking](add link to PoS page){target=\_blank}, [accounts](add a link to accounts pages){target=\_blank}, balances, and [governance](add a link to govenance page){target=\_blank}.

#### Agile Coretime

The new and more efficient way of obtaining core on Polkadot is to go through the process of purchasing coretime.

> [Agile coretime](add link to Agile Coretime page) improves the efficient use of Polkadot's network resources and offers economic flexibility for developers, extending Polkadot's capabilities far beyond the original vision outlined in the [whitepaper](https://polkadot.com/papers/Polkadot-whitepaper.pdf){target=\_blank}.

Reviewer 2: ```suggestion
[Agile coretime](todo: add link to Agile Coretime page) improves the efficient use of Polkadot's network resources and offers economic flexibility for developers, extending Polkadot's capabilities far beyond the original vision outlined in the [whitepaper](https://polkadot.com/papers/Polkadot-whitepaper.pdf){target=\_blank}.
```

### Polkadot's Resilience

Decentralization is a vital component of blockchain networks, but it comes with trade-offs:

- An overly decentralized network may face challenges in reaching consensus and require significant energy to operate
- Also, a network that achieves consensus quickly risks centralization, making it easier to manipulate or attack

A network should be decentralized enough to prevent manipulative or malicious influence. In this sense, decentralization is a tool for achieving resilience. 

Polkadot 1.0 currently achieves resilience through several strategies:

- **Nominated Proof of Stake (NPoS)** - this ensures that the stake per validator is maximized and evenly distributed among validators

- **Decentralized nodes** - designed to encourage operators to join the network. This program aims to expand and diversify the validators in the ecosystem who aim to become independent of the program during their term. Feel free to explore more about the program on the official [Decentralized Nodes](https://nodes.web3.foundation/){target=\_blank} page

> - **On-chain treasury and governance** - known as [OpenGov](link to opengov overview page){target=\_blank}, this system allows every decision to be made through public referenda, enabling any token holder to cast a vote

Reviewer 2: ```suggestion
- **On-chain treasury and governance** - known as [OpenGov](todo: link to opengov overview page){target=\_blank}, this system allows every decision to be made through public referenda, enabling any token holder to cast a vote
```

- **Availability** - indicates how easily users can access blockspace. It should be easily accessible, allowing diverse business models to thrive, ideally regulated by a marketplace based on demand and supplemented by options for "second-hand" blockspace

Polkadot is built on core blockspace principles, but there's room for improvement. Tasks like balance transfers, staking, and governance are managed on the relay chain. 

Delegating these responsibilities to [system chains](add link to System Chains section) could enhance flexibility and allow the relay chain to concentrate on providing shared security and interoperability.

!!!note
    For more information about blockspace, watch [Robert Habermeier’s interview](https://www.youtube.com/watch?v=e1vISppPwe4){target=\_blank} or read his [technical blog post](https://www.rob.tech/blog/polkadot-blockspace-over-blockchains/){target=\_blank}.

## DOT Token

DOT is the native token of the Polkadot network, much like BTC for Bitcoin and Ether for the Ethereum blockchain. DOT has 10 decimals, uses the Planck base unit, and has a balance type of `u128`. The same is true for Kusama's KSM token with the exception of having 12 decimals.

> ??? info "Redomination of DOT"

Reviewer 2: ```suggestion
??? info "Redenomination of DOT"
```

---
title: Overview
description: Explore the fundamental concepts of the Polkadot protocol, including account addresses, extrinsics, events and runtime upgrades.
---

# Overview

> Since the release of Bitcoin in 2009, blockchain projects increased exponentially to the order of tens of thousands. 

Reviewer 3: I'm completely sure we should leave this.. I thin starting starting with the next paragraph should suffice.
```suggestion
Since the release of Bitcoin in 2009, the amount of blockchain projects increased to the order of tens of thousands. 
```

> Different projects have different value propositions, suggesting that the future will be multi-chain and that inter-chain communication will be crucial to establish collaborations and leveraging each other strengths.

Reviewer 3: Just my own input.. not even multi-chain, but rather multi-_protocol_, but probably better to keep multi-chain for the sake of messaging ;)

## Polkadot 1.0

Polkadot 1.0 represents the state of Polkadot as of 2023, coinciding with the release of [Polkadot runtime v1.0.0](https://github.com/paritytech/polkadot/releases/tag/v1.0.0){target=\_blank}. This section will focus on Polkadot 1.0, along with philosophical insights into network resilience and blockspace.

As a Layer-0 blockchain, Polkadot contributes to the multi-chain vision through several key innovations and initiatives, including:

1. **Application-specific Layer-1 blockchains (parachains)** - Polkadot's sharded network allows for parallel transaction processing, with shards that can have unique state transition functions, enabling custom-built L1 chains optimized for specific applications

> 2. **Shared security and scalability** - L1 chains connected to Polkadot benefit from its [Nominated-Proof-of-Stake (NPoS)](link to the PoS page) system, providing security out-of-the-box without the need to bootstrap their own

Reviewer 3: We can probably link the wiki section here: https://wiki.polkadot.network/docs/learn-consensus#nominated-proof-of-stake

> 3. **Secure interoperability** - Polkadot's native interoperability enables L1 and L2 chains to seamlessly exchange data and value across parachains

Reviewer 3: My wording is not so great here.. but you get the idea
```suggestion
3. **Secure interoperability** - Polkadot's native interoperability enables seamless data and value exchange between parachains. This interoperability can also be used outside of the ecosystem for bridging with external networks
```

4. **Resilient infrastructure** - decentralized and scalable, Polkadot ensures ongoing support for development and community initiatives via its on-chain [treasury](https://wiki.polkadot.network/docs/learn-polkadot-opengov-treasury){target=\_blank} and governance

5. **Rapid L1 development** - the [Polkadot SDK](link to Polkadot SDK page in polkadot protocol section) allows fast, flexible creation and deployment of Layer-1 chains

6. **Cultivating the next generation of Web3 developers** - Polkadot supports the growth of Web3 core developers through initiatives such as:

    - [Polkadot Blockchain Academy](https://polkadot.com/blockchain-academy){target=\_blank}
    - [Polkadot Alpha Program](https://polkadot.com/alpha-program){target=\_blank}

>     - [Polkadot Developer Heroes Program](https://wiki.polkadot.network/docs/dev-heroes){target=\_blank}

Reviewer 3: Is Heroes still active? Not sure it is.. need to check for both here and the Wiki

    - [EdX Courses](https://www.edx.org/school/web3x){target=\_blank}
    - Rust and Substrate Courses (coming soon)

### High Level Architecture

Polkadot features a chain that serves as the central component of the system. This chain is depicted as a ring, encircled by several parachains that are connected to it. 

> According to Polkadot's design, any chain that can compile to WebAssembly (Wasm) and complies with the  chain's API can integrate into the Polkadot network as a parachain.

Reviewer 3: Parachains/ELVES protocol should be linked here. It could also be useful to mention Substrate/Polkadot SDK as the means most people use to build these chains.

```suggestion
According to Polkadot's design, any blockchain that can compile to WebAssembly (Wasm) and adheres to the Parachains Protocol become a parachain on the Polkadot network.
```

### Polkadot's Additional Functionalities

The Polkadot  chain oversees crowdloans and auctions. Chain cores were leased through auctions for three-month periods, up to a maximum of two years. 

Crowdloans enabled users to securely lend funds to teams for lease deposits in exchange for pre-sale tokens, which is the only way to access slots on Polkadot 1.0.

!!!note
    Auction are deprecated in favor of Coretime

Additionally, the  chain handles [staking](add link to PoS page){target=\_blank}, [accounts](add a link to accounts pages){target=\_blank}, balances, and [governance](add a link to govenance page){target=\_blank}.

#### Agile Coretime

The new and more efficient way of obtaining core on Polkadot is to go through process of purchasing a **coretime**.

[**Agile Coretime**](add link to Agile Coretime page) improves the efficient use of Polkadot's network resources and offers economic flexibility for developers, extending Polkadot's capabilities far beyond the original vision outlined in its [whitepaper](https://polkadot.com/papers/Polkadot-whitepaper.pdf){target=\_blank}.

> It enables parachains to purchase monthly "bulk" allocations of **coretime**(The time allocated for utilizing a core, measured in Polkadot relay chain blocks), ensuring heavy-duty parachains that need to author a block every 12 seconds (or every 6 seconds with [Asynchronous Backing](add link to async backing page)) can reliably renew their coretime each month. 

Reviewer 3: I think async backing should be the default now, but we need to check

- **Security** - measures the robustness of blockspace in Proof-of-Stake (PoS) networks, linked to the stake locked on validator nodes, the variance in stake among validators, and the total number of validators. It also considers social centralization (how many validators are owned by single operators) and physical centralization (how many validators run on the same service provider)

- **Flexibility** - reflects the functionalities and types of data that can be stored, with high-quality data essential to avoid bottlenecks in critical processes

- **Availability** - indicates how easily users can access blockspace. It should be easily accessible, allowing diverse business models to thrive, ideally regulated by a marketplace based on demand and supplemented by options for "second-hand" blockspace

Polkadot is built on core blockspace principles, but there’s room for improvement. Currently, tasks like balance transfers, staking, and governance are managed on the relay chain. 

Delegating these responsibilities to [system chains](add link to System Chains section) could enhance flexibility and allow the relay chain to concentrate on providing shared security and interoperability.

> Currently, blockspace is accessible only through auctions, and the auction winner gains access to a "freighter of blocks," whether needed or not. This setup creates high entry barriers and can lead to wasted energy and resources.

Reviewer 3: Probably best to remove this paragraph, as this is no longer true due to Coretime

Importantly, this did not affect the total number of base units (plancks) in the network, only how a single DOT is represented.

The redenomination became effective 72 hours after transfers were enabled, occurring at block **1,248,328** on 21 August 2020 around 16:50 UTC.

## The Planck Unit

The smallest unit of account balance on Substrate-based blockchains (such as Polkadot and Kusama) is called **Planck**, named after the **Planck Length**, the smallest measurable distance in the physical universe. 

Similar to how BTC's smallest unit is the **Satoshi** and ETH's is the **Wei**, Polkadot’s native token DOT is equal to 1e10 Planck, while Kusama’s native token KSM equals 1e12 Planck.

## What are the uses of DOT?

DOT serves four primary functions within the Polkadot network:

1. **Governance** - it is used to participate in the governance of the network
2. **Staking** - DOT is staked to support the network’s operation and security
3. **Bonding** - it is bonded to connect new chains to Polkadot as parachains
4. **Buying Coretime** - used to purchase coretime in-bulk or on-demand and access the  chain to benefit from Polkadot's security and interoperability

> Additionally, DOT can serve as a transferable token. For example, DOT held in the Treasury can be allocated to teams developing projects that benefit the Polkadot ecosystem.

Reviewer 3: Maybe we can add a "What's Next" page similar to other pages?

---
title: Overview
description: Explore the fundamental concepts of the Polkadot protocol, including account addresses, extrinsics, events and runtime upgrades.
---

# Overview

> 

Reviewer 2: Can you please add an ## Introduction section? it should include what the page is about with a generic like "This guide will...", "In this guide, you'll learn...", etc.

Reviewer 4: Added Intro

> Since the release of Bitcoin in 2009, the amount of blockchain projects increased to the order of tens of thousands. 

Reviewer 2: ```suggestion
Since the release of Bitcoin in 2009, the amount of blockchain projects has increased to the order of tens of thousands. 
```

> Different projects have different value propositions, suggesting that the future will be multi-chain and that inter-chain communication will be crucial to establish collaborations and leveraging each other strengths.

Reviewer 2: ```suggestion
Different projects have different value propositions, suggesting that the future will be multi-chain and that inter-chain communication will be crucial to establishing collaborations and leveraging each other strengths.
```

## Polkadot 1.0

Polkadot 1.0 represents the state of Polkadot as of 2023, coinciding with the release of [Polkadot runtime v1.0.0](https://github.com/paritytech/polkadot/releases/tag/v1.0.0){target=\_blank}. This section will focus on Polkadot 1.0, along with philosophical insights into network resilience and blockspace.

As a Layer-0 blockchain, Polkadot contributes to the multi-chain vision through several key innovations and initiatives, including:

> 1. **Application-specific Layer-1 blockchains (parachains)** - Polkadot's sharded network allows for parallel transaction processing, with shards that can have unique state transition functions, enabling custom-built L1 chains optimized for specific applications

Reviewer 2: These don't need to be in sequential order, so this list can be turned into an unordered list (using `-` instead)

3. **Secure interoperability** - Polkadot's native interoperability enables seamless data and value exchange between parachains. This interoperability can also be used outside of the ecosystem for bridging with external networks

4. **Resilient infrastructure** - decentralized and scalable, Polkadot ensures ongoing support for development and community initiatives via its on-chain [treasury](https://wiki.polkadot.network/docs/learn-polkadot-opengov-treasury){target=\_blank} and governance

5. **Rapid L1 development** - the [Polkadot SDK](link to Polkadot SDK page in polkadot protocol section) allows fast, flexible creation and deployment of Layer-1 chains

6. **Cultivating the next generation of Web3 developers** - Polkadot supports the growth of Web3 core developers through initiatives such as:

    - [Polkadot Blockchain Academy](https://polkadot.com/blockchain-academy){target=\_blank}
    - [Polkadot Alpha Program](https://polkadot.com/alpha-program){target=\_blank}

>     - [EdX Courses](https://www.edx.org/school/web3x){target=\_blank}

Reviewer 2: I don't think courses needs to be capitalized here

>     - Rust and Substrate Courses (coming soon)

Reviewer 2: Is this an official name? Otherwise, we don't need to capitalize courses

### High Level Architecture

> Polkadot features a chain that serves as the central component of the system. This chain is depicted as a ring, encircled by several parachains that are connected to it. 

Reviewer 2: ```suggestion
Polkadot features a chain that serves as the central component of the system. This chain is depicted as a ring encircled by several parachains that are connected to it. 
```

> According to Polkadot's design, any blockchain that can compile to WebAssembly (Wasm) and adheres to the Parachains Protocol become a parachain on the Polkadot network.

Reviewer 2: Is Parachains Protocol something specific? Otherwise, don't need to capitalize

Reviewer 3: It's usually capitalized; although it might be being deprecated in favor of ELVES, @filippoweb3 would appreciate some clarity here!

> According to Polkadot's design, any blockchain that can compile to WebAssembly (Wasm) and adheres to the Parachains Protocol become a parachain on the Polkadot network.

Reviewer 2: ```suggestion
According to Polkadot's design, any blockchain that can compile to WebAssembly (Wasm) and adheres to the Parachains Protocol becomes a parachain on the Polkadot network.
```

> Here’s a high-level overview of the Polkadot protocol architecture

Reviewer 2: ```suggestion
Here’s a high-level overview of the Polkadot protocol architecture:
```

> ![](/images/polkadot-protocol/architecture/polkadot-chain/overview/relay-chain-architecture.webp)

Reviewer 2: This image should follow the `filename-1.webp` naming convention, so `overview-1.webp`

> Parachains propose blocks to the Polkadot validators, who perform checks for availability and validity before finalizing them. With the  chain providing security, collators — full nodes of Parachains — can focus on their tasks without needing strong incentives.

Reviewer 2: ```suggestion
Parachains propose blocks to Polkadot validators, who check for availability and validity before finalizing them. With the chain providing security, collators—full nodes of parachains—can focus on their tasks without needing strong incentives.
```

> Parachains propose blocks to the Polkadot validators, who perform checks for availability and validity before finalizing them. With the  chain providing security, collators — full nodes of Parachains — can focus on their tasks without needing strong incentives.

Reviewer 3: I would specify the relay chain:

"With the **relay** chain providing security"

> The [Cross-Consensus Messaging Format (XCM)](add link to XCM page){target=\_blank} allows Parachains to exchange messages freely, leveraging the  chain's security for trust-free communication.

Reviewer 2: ```suggestion
The [Cross-Consensus Messaging Format (XCM)](add link to XCM page){target=\_blank} allows parachains to exchange messages freely, leveraging the chain's security for trust-free communication.
```

In order to interact with chains that want to use their own finalization process (e.g. Bitcoin), Polkadot has [bridges](add link to bridges pages, Snowbridge, etc){target=\_blank}  that offer two-way compatibility, meaning that transactions can be made between different parachains.

### Polkadot's Additional Functionalities

> The Polkadot  chain oversees crowdloans and auctions. Chain cores were leased through auctions for three-month periods, up to a maximum of two years. 

Reviewer 2: ```suggestion
The Polkadot chain oversees crowdloans and auctions. Chain cores were leased through auctions for three-month periods, up to a maximum of two years. 
```

> The Polkadot  chain oversees crowdloans and auctions. Chain cores were leased through auctions for three-month periods, up to a maximum of two years. 

Reviewer 3: Is there something new regarding how we are saying "the Polkadot chain" versus specifying the relay chain?

Reviewer 2: No, it should be the relay chain

Crowdloans enabled users to securely lend funds to teams for lease deposits in exchange for pre-sale tokens, which is the only way to access slots on Polkadot 1.0.

!!!note

>     Auction are deprecated in favor of Coretime

Reviewer 2: ```suggestion
    Auctions are deprecated in favor of [coretime](todo: update link).
```

> Additionally, the  chain handles [staking](add link to PoS page){target=\_blank}, [accounts](add a link to accounts pages){target=\_blank}, balances, and [governance](add a link to govenance page){target=\_blank}.

Reviewer 2: ```suggestion
Additionally, the chain handles [staking](add link to PoS page){target=\_blank}, [accounts](add a link to accounts pages){target=\_blank}, balances, and [governance](add a link to govenance page){target=\_blank}.
```

#### Agile Coretime

> The new and more efficient way of obtaining core on Polkadot is to go through process of purchasing a **coretime**.

Reviewer 2: ```suggestion
The new and more efficient way of obtaining core on Polkadot is to go through the process of purchasing coretime.
```

> [**Agile Coretime**](add link to Agile Coretime page) improves the efficient use of Polkadot's network resources and offers economic flexibility for developers, extending Polkadot's capabilities far beyond the original vision outlined in its [whitepaper](https://polkadot.com/papers/Polkadot-whitepaper.pdf){target=\_blank}.

Reviewer 2: ```suggestion
[Agile coretime](add link to Agile Coretime page) improves the efficient use of Polkadot's network resources and offers economic flexibility for developers, extending Polkadot's capabilities far beyond the original vision outlined in the [whitepaper](https://polkadot.com/papers/Polkadot-whitepaper.pdf){target=\_blank}.
```

> It enables parachains to purchase monthly "bulk" allocations of **coretime**(The time allocated for utilizing a core, measured in Polkadot relay chain blocks), ensuring heavy-duty parachains that can author a block every 6 seconds with [Asynchronous Backing](add link to async backing page) can reliably renew their coretime each month. Although six second block times is now the default, parachains have the option of producing blocks less frequently. 

Reviewer 2: ```suggestion
It enables parachains to purchase monthly "bulk" allocations of coretime (the time allocated for utilizing a core, measured in Polkadot relay chain blocks), ensuring heavy-duty parachains that can author a block every six seconds with [Asynchronous Backing](add link to async backing page) can reliably renew their coretime each month. Although six-second block times are now the default, parachains have the option of producing blocks less frequently. 
```

> Renewal orders are prioritized over new orders, offering stability against price fluctuations and helping parachains budget more effectively for their project costs.

Reviewer 2: ```suggestion
Renewal orders are prioritized over new orders, offering stability against price fluctuations and helping parachains budget more effectively for project costs.
```

### Polkadot's Resilience

> Decentralization is a vital component of blockchain networks, but it comes with trade-offs:
> 
> - An overly decentralized network may face challenges in reaching consensus and require significant energy to operate
> - Also a network that achieves consensus quickly risks centralization, making it easier to manipulate or attack
> 
> Ideally, a network should be decentralized enough to prevent any manipulative or malicious influence. In this sense, decentralization serves as a tool to achieve resilience. 
> 
> **Polkadot 1.0** currently achieves resilience through several strategies:
> 
> 1. **Nominated Proof of Staking (NPoS)** - this ensures that the stake per validator is maximized and evenly distributed among validators
> 
> 2. **Decentralized Nodes** - designed to encourage operators to join the network, this program aims to expand and diversify the validators in the ecosystem that aim to become independent of the program during their term. Feel free to explore more about the program on the official [Decentralized Nodes](https://nodes.web3.foundation/){target=\_blank} page.
> 
> 3. **On-Chain Treasury and Governance** - known as [OpenGov](link to opengov overview page){target=\_blank}, this system allows every decision to be made through public referenda, enabling any token holder to cast a vote

Reviewer 2: ```suggestion
Decentralization is a vital component of blockchain networks, but it comes with trade-offs:

- An overly decentralized network may face challenges in reaching consensus and require significant energy to operate
- Also, a network that achieves consensus quickly risks centralization, making it easier to manipulate or attack

A network should be decentralized enough to prevent manipulative or malicious influence. In this sense, decentralization is a tool for achieving resilience. 

Polkadot 1.0 currently achieves resilience through several strategies:

- **Nominated Proof of Stake (NPoS)** - this ensures that the stake per validator is maximized and evenly distributed among validators

- **Decentralized nodes** - designed to encourage operators to join the network. This program aims to expand and diversify the validators in the ecosystem who aim to become independent of the program during their term. Feel free to explore more about the program on the official [Decentralized Nodes](https://nodes.web3.foundation/){target=\_blank} page

- **On-chain treasury and governance** - known as [OpenGov](link to opengov overview page){target=\_blank}, this system allows every decision to be made through public referenda, enabling any token holder to cast a vote
```

### Polkadot's Blockspace

Polkadot 1.0’s design allows for the commoditization of blockspace.

> Blockspace is the capacity of a blockchain to finalize and commit operations, encompassing its security, computing, and storage capabilities. Its characteristics can vary across different blockchains, affecting **security**, **flexibility**, and **availability**.
> 
> - **Security** - measures the robustness of blockspace in Proof-of-Stake (PoS) networks, linked to the stake locked on validator nodes, the variance in stake among validators, and the total number of validators. It also considers social centralization (how many validators are owned by single operators) and physical centralization (how many validators run on the same service provider)
> 
> - **Flexibility** - reflects the functionalities and types of data that can be stored, with high-quality data essential to avoid bottlenecks in critical processes
> 
> - **Availability** - indicates how easily users can access blockspace. It should be easily accessible, allowing diverse business models to thrive, ideally regulated by a marketplace based on demand and supplemented by options for "second-hand" blockspace
> 
> Polkadot is built on core blockspace principles, but there’s room for improvement. Currently, tasks like balance transfers, staking, and governance are managed on the relay chain. 
> 
> Delegating these responsibilities to [system chains](add link to System Chains section) could enhance flexibility and allow the relay chain to concentrate on providing shared security and interoperability.

Reviewer 2: ```suggestion
Blockspace is a blockchain's capacity to finalize and commit operations, encompassing its security, computing, and storage capabilities. Its characteristics can vary across different blockchains, affecting security, flexibility, and availability.

- **Security** - measures the robustness of blockspace in Proof of Stake (PoS) networks linked to the stake locked on validator nodes, the variance in stake among validators, and the total number of validators. It also considers social centralization (how many validators are owned by single operators) and physical centralization (how many validators run on the same service provider)

- **Flexibility** - reflects the functionalities and types of data that can be stored, with high-quality data essential to avoid bottlenecks in critical processes

- **Availability** - indicates how easily users can access blockspace. It should be easily accessible, allowing diverse business models to thrive, ideally regulated by a marketplace based on demand and supplemented by options for "second-hand" blockspace

Polkadot is built on core blockspace principles, but there's room for improvement. Tasks like balance transfers, staking, and governance are managed on the relay chain. 

Delegating these responsibilities to [system chains](add link to System Chains section) could enhance flexibility and allow the relay chain to concentrate on providing shared security and interoperability.
```

!!!note

>     For more information about blockspace see this [interview](https://www.youtube.com/watch?v=e1vISppPwe4){target=\_blank} with **Robert Habermeier** as well as this [article](https://www.rob.tech/blog/polkadot-blockspace-over-blockchains/){target=\_blank} by him.

Reviewer 2: ```suggestion
    For more information about blockspace, watch [Robert Habermeier’s interview](https://www.youtube.com/watch?v=e1vISppPwe4){target=\_blank} or read his [technical blog post](https://www.rob.tech/blog/polkadot-blockspace-over-blockchains/){target=\_blank}.
```

However, if you prefer to bypass these tools and deal directly with the raw chain data, Polkadot uses the [SCALE codec](https://github.com/paritytech/parity-scale-codec#parity-scale-codec){target=\_blank} to encode block and transaction data. 

This allows you to implement your own codecs for custom interactions with the chain if needed.

### Runtime Upgrades

Runtime upgrades allow Polkadot to change the logic of the chain without the need for a hard fork. 

!!!note
    You can find a guide for how to properly perform a runtime upgrade [here](runtime upgrades page).

#### Runtime Versioning

There are a number of fields that are a part of the overall `RuntimeVersion`.

Apart the `runtime_version` there is also the `transaction_version` which denotes how to correctly encode/decode calls for a given runtime (useful for hardware wallets). 

The reason `transaction_version` is separate from `runtime_version` is that it explicitly notes that the call interface is broken/not compatible.

## DOT Token

### Tokens

- Token decimals:
    - Polkadot (DOT): 10
    - Kusama (KSM): 12
- Base unit: "Planck"
- Balance type: `u128`

### What is DOT?

> DOT is the native token of the Polkadot network, much like BTC is for Bitcoin and Ether is for the Ethereum blockchain.

Reviewer 2: ```suggestion
DOT is the native token of the Polkadot network, much like BTC for Bitcoin and Ether for the Ethereum blockchain.
```

> ### Redomination

Reviewer 2: ```suggestion
### Redenomination
```

> Polkadot conducted a community poll, which ended on **27 July 2020** at block **888,888** to decide whether to redenominate the DOT token. 
> 
> The stakeholders chose to redenominate the token, changing the value of 1 DOT from 1e12 plancks to 1e10 plancks. 
> 
> Importantly, this did not affect the total number of base units (plancks) in the network, only how a single DOT is represented.
> 
> The redenomination became effective 72 hours after transfers were enabled, occurring at block **1,248,328** on 21 August 2020 around 16:50 UTC.

Reviewer 2: ```suggestion
Polkadot conducted a community poll, which ended on 27 July 2020 at block 888,888, to decide whether to redenominate the DOT token. 

The stakeholders chose to redenominate the token, changing the value of 1 DOT from 1e12 plancks to 1e10 plancks. 

Importantly, this did not affect the network's total number of base units (plancks); it only affects how a single DOT is represented.

The redenomination became effective 72 hours after transfers were enabled, occurring at block 1,248,328 on 21 August 2020 around 16:50 UTC.
```

## The Planck Unit

> The smallest unit of account balance on Substrate-based blockchains (such as Polkadot and Kusama) is called **Planck**, named after the **Planck Length**, the smallest measurable distance in the physical universe. 
> 
> Similar to how BTC's smallest unit is the **Satoshi** and ETH's is the **Wei**, Polkadot’s native token DOT is equal to 1e10 Planck, while Kusama’s native token KSM equals 1e12 Planck.

Reviewer 2: ```suggestion
The smallest unit of account balance on Substrate-based blockchains (such as Polkadot and Kusama) is called _Planck_, named after the Planck length, the smallest measurable distance in the physical universe. 

Similar to how BTC's smallest unit is the Satoshi and ETH's is the Wei, Polkadot's native token DOT equals 1e10 Planck, while Kusama's native token KSM equals 1e12 Planck.
```

> ## What are the uses of DOT?

Reviewer 2: ```suggestion
## What Are the Uses of DOT?
```

> DOT serves three primary functions within the Polkadot network:
> 
> 1. **Governance** - it is used to participate in the governance of the network
> 2. **Staking** - DOT is staked to support the network’s operation and security
> 3. **Buying Coretime** - used to purchase coretime in-bulk or on-demand and access the  chain to benefit from Polkadot's security and interoperability
> 
> Additionally, DOT can serve as a transferable token. For example, DOT held in the Treasury can be allocated to teams developing projects that benefit the Polkadot ecosystem.

Reviewer 2: ```suggestion
DOT serves three primary functions within the Polkadot network:

- **Governance** - it is used to participate in the governance of the network
- **Staking** - DOT is staked to support the network's operation and security
- **Buying coretime** - used to purchase coretime in-bulk or on-demand and access the  chain to benefit from Polkadot's security and interoperability

Additionally, DOT can serve as a transferable token. For example, DOT, held in the treasury, can be allocated to teams developing projects that benefit the Polkadot ecosystem.
```

## JAM and the Road Ahead

The Join-Accumulate Machine (JAM) represents a transformative redesign of Polkadot's core architecture, envisioned as the successor to the current relay chain. Unlike traditional blockchain architectures, JAM introduces a unique computational model that processes work through two primary functions: Join (which handles the integration of data) and Accumulate (which folds computations into the chain's state). 

> Operating as a domain-specific rollup chain, JAM removes many of the opinions and constraints of the current relay chain while maintaining its core security properties. It's designed to be more generic and flexible, allowing for permissionless code execution through "services" that can be deployed without governance approval. What makes JAM particularly innovative is its transactionless nature and its efficient pipeline processing model, which places the prior state root in block headers instead of the posterior state root, enabling more effective utilization of block time for computations. This architectural evolution promises to enhance Polkadot's scalability and flexibility while maintaining its robust security guarantees.

Reviewer 3: I am not sure this is entirely accurate, as can do more than that, I think we can just start the sentence straight away.
```suggestion
JAM removes many of the opinions and constraints of the current relay chain while maintaining its core security properties. It's designed to be more generic and flexible, allowing for permissionless code execution through "services" that can be deployed without governance approval. What makes JAM particularly innovative is its transactionless nature and its efficient pipeline processing model, which places the prior state root in block headers instead of the posterior state root, enabling more effective utilization of block time for computations. This architectural evolution promises to enhance Polkadot's scalability and flexibility while maintaining its robust security guarantees.
```

> The Join-Accumulate Machine (JAM) represents a transformative redesign of Polkadot's core architecture, envisioned as the successor to the current relay chain. Unlike traditional blockchain architectures, JAM introduces a unique computational model that processes work through two primary functions: Join (which handles the integration of data) and Accumulate (which folds computations into the chain's state). 
> 
> Operating as a domain-specific rollup chain, JAM removes many of the opinions and constraints of the current relay chain while maintaining its core security properties. It's designed to be more generic and flexible, allowing for permissionless code execution through "services" that can be deployed without governance approval. What makes JAM particularly innovative is its transactionless nature and its efficient pipeline processing model, which places the prior state root in block headers instead of the posterior state root, enabling more effective utilization of block time for computations. This architectural evolution promises to enhance Polkadot's scalability and flexibility while maintaining its robust security guarantees.
> 
> Also of note, JAM is planned to be rolled out to Polkadot as a single, complete upgrade, rather than a steady stream of smaller updates, which can have a tendency to increase developer overhead in addressing breaking changes. 

Reviewer 2: ```suggestion
The Join-Accumulate Machine (JAM) represents a transformative redesign of Polkadot's core architecture, envisioned as the successor to the current relay chain. Unlike traditional blockchain architectures, JAM introduces a unique computational model that processes work through two primary functions: join (which handles data integration) and accumulate (which folds computations into the chain's state). 

Operating as a domain-specific rollup chain, JAM removes many of the opinions and constraints of the current relay chain while maintaining its core security properties. It's designed to be more generic and flexible, allowing for permissionless code execution through "services" that can be deployed without governance approval. What makes JAM particularly innovative is its transactionless nature and efficient pipeline processing model, which places the prior state root in block headers instead of the posterior state root, enabling more effective utilization of block time for computations. This architectural evolution promises to enhance Polkadot's scalability and flexibility while maintaining robust security guarantees.

Also of note, JAM is planned to be rolled out to Polkadot as a single, complete upgrade rather than a steady stream of smaller updates, which can tend to increase developer overhead in addressing breaking changes.
```

---
title: Overview
description: Explore the fundamental concepts of the Polkadot protocol, including account addresses, extrinsics, events and runtime upgrades.
---

# Overview

Since the release of Bitcoin in 2009, blockchain projects increased exponentially to the order of tens of thousands. 

Different projects have different value propositions, suggesting that the future will be multi-chain and that inter-chain communication will be crucial to establish collaborations and leveraging each other strengths.

## Polkadot 1.0

Polkadot 1.0 represents the state of Polkadot as of 2023, coinciding with the release of [Polkadot runtime v1.0.0](https://github.com/paritytech/polkadot/releases/tag/v1.0.0){target=\_blank}. This section will focus on Polkadot 1.0, along with philosophical insights into network resilience and blockspace.

As a Layer-0 blockchain, Polkadot contributes to the multi-chain vision through several key innovations and initiatives, including:

> 1. **Application-specific Layer-1 blockchains (Polkadot chains)**: Polkadot's sharded network allows for parallel transaction processing, with shards that can have unique state transition functions, enabling custom-built L1 chains optimized for specific applications.
> 
> 2. **Shared security and scalability**: L1 chains connected to Polkadot benefit from its [Nominated-Proof-of-Stake (NPoS)](link to the PoS page) system, providing security out-of-the-box without the need to bootstrap their own.
> 
> 3. **Secure interoperability**: Polkadot's native interoperability enables L1 and L2 chains to seamlessly exchange data and value across Polkadot chains.
> 
> 4. **Resilient infrastructure**: Decentralized and scalable, Polkadot ensures ongoing support for development and community initiatives via its on-chain [treasury](https://wiki.polkadot.network/docs/learn-polkadot-opengov-treasury){target=\_blank} and governance.
> 
> 5. **Rapid L1 development**: The [Polkadot SDK](link to Polkadot SDK page in polkadot protocol section) allows fast, flexible creation and deployment of Layer-1 chains.
> 
> 6. **Cultivating the next generation of Web3 developers**: Polkadot supports the growth of Web3 core developers through initiatives such as:

Reviewer: ```suggestion
1. **Application-specific Layer-1 blockchains (Polkadot chains)** - Polkadot's sharded network allows for parallel transaction processing, with shards that can have unique state transition functions, enabling custom-built L1 chains optimized for specific applications

2. **Shared security and scalability** - L1 chains connected to Polkadot benefit from its [Nominated-Proof-of-Stake (NPoS)](link to the PoS page) system, providing security out-of-the-box without the need to bootstrap their own

3. **Secure interoperability** - Polkadot's native interoperability enables L1 and L2 chains to seamlessly exchange data and value across Polkadot chains

4. **Resilient infrastructure** - decentralized and scalable, Polkadot ensures ongoing support for development and community initiatives via its on-chain [treasury](https://wiki.polkadot.network/docs/learn-polkadot-opengov-treasury){target=\_blank} and governance

5. **Rapid L1 development** - the [Polkadot SDK](link to Polkadot SDK page in polkadot protocol section) allows fast, flexible creation and deployment of Layer-1 chains

6. **Cultivating the next generation of Web3 developers** - Polkadot supports the growth of Web3 core developers through initiatives such as:
```

    - [Polkadot Blockchain Academy](https://polkadot.com/blockchain-academy){target=\_blank}
    - [Polkadot Alpha Program](https://polkadot.com/alpha-program){target=\_blank}
    - [Polkadot Developer Heroes Program](https://wiki.polkadot.network/docs/dev-heroes){target=\_blank}
    - [EdX Courses](https://www.edx.org/school/web3x){target=\_blank}
    - Rust and Substrate Courses (coming soon)

### High Level Architecture

Polkadot features a chain that serves as the central component of the system. This chain is depicted as a ring, encircled by several Polkadot chains that are connected to it. 

According to Polkadot's design, any chain that can compile to WebAssembly (Wasm) and complies with the  chain's API can integrate into the Polkadot network as a parachain.

Here’s a high-level overview of the Polkadot protocol architecture

![](/images/polkadot-protocol/architecture/polkadot-chain/overview/relay-chain-architecture.webp)

> Polkadot chains propose blocks to the Polkadot validators, who perform checks for availability and validity before finalizing them. With the  chain providing security, collators — full nodes of Polkadot chains — can focus on their tasks without needing strong incentives.

Reviewer: Do we want to use the term `Polkadot chains` or maybe something like appchains / parachains ?

Reviewer 5: Whats the consensus on potentially using the term 'Rollups'? Imo it's important to keep it consistent throughout the doc's so I'd be happy with with whatever we've decided upon in other PR's.

Author: @RemyLeBerre, for now we decided to keep the word "parachains" until "rollup" is officialized

The [Cross-Consensus Messaging Format (XCM)](add link to XCM page){target=\_blank} allows Polkadot chains to exchange messages freely, leveraging the  chain's security for trust-free communication.

In order to interact with chains that want to use their own finalization process (e.g. Bitcoin), Polkadot has [bridges](add link to bridges pages, Snowbridge, etc){target=\_blank}  that offer two-way compatibility, meaning that transactions can be made between different Polkadot chains.

### Polkadot's Additional Functionalities

The Polkadot  chain oversees crowdloans and auctions. Chain cores were leased through auctions for three-month periods, up to a maximum of two years. 

Crowdloans enabled users to securely lend funds to teams for lease deposits in exchange for pre-sale tokens, which is the only way to access slots on Polkadot 1.0.

!!!note
    Auction are deprecated in favor of Coretime

> Additionally, the  chain handles [staking](add link to PoS page){target=\_blank}, [accounts](add a link to accounts pages){target=\_blank}, balances, and [governance](add a link to govenance page){target=\_blank}.

Reviewer: I would write a brief explanation of coretime first because it is the current mechanism being used, and then you can add crowdloans and auctions as a note

Author: Explanation added

### Polkadot's Resilience

Decentralization is a vital component of blockchain networks, but it comes with trade-offs:

> - An overly decentralized network may face challenges in reaching consensus and require significant energy to operate.
> - Also a network that achieves consensus quickly risks centralization, making it easier to manipulate or attack.

Reviewer: ```suggestion
- An overly decentralized network may face challenges in reaching consensus and require significant energy to operate
- Also a network that achieves consensus quickly risks centralization, making it easier to manipulate or attack
```

Ideally, a network should be decentralized enough to prevent any manipulative or malicious influence. In this sense, decentralization serves as a tool to achieve resilience. 

**Polkadot 1.0** currently achieves resilience through several strategies:

1. **Nominated Proof of Staking (NPoS)**: This ensures that the stake per validator is maximized and evenly distributed among validators.

> 2. **1KV Program**: Designed to encourage new operators to join the network, this program enhances both physical decentralization (the number of validator nodes per service provider) and social decentralization (the number of validator nodes per operator). Feel free to explore these metrics using the [Polkawatch App](https://polkadot.polkawatch.app/){target=\_blank}.

Reviewer: This program is being sunset. The new one is [Decentralized Nodes](https://nodes.web3.foundation/)

Author: updated

> 1. **Nominated Proof of Staking (NPoS)**: This ensures that the stake per validator is maximized and evenly distributed among validators.
> 
> 2. **1KV Program**: Designed to encourage new operators to join the network, this program enhances both physical decentralization (the number of validator nodes per service provider) and social decentralization (the number of validator nodes per operator). Feel free to explore these metrics using the [Polkawatch App](https://polkadot.polkawatch.app/){target=\_blank}.
> 
> 3. **On-Chain Treasury and Governance**: Known as [OpenGov](link to opengov overview page){target=\_blank}, this system allows every decision to be made through public referenda, enabling any token holder to cast a vote.

Reviewer: ```suggestion
1. **Nominated Proof of Staking (NPoS)** - this ensures that the stake per validator is maximized and evenly distributed among validators

2. **1KV Program** - designed to encourage new operators to join the network, this program enhances both physical decentralization (the number of validator nodes per service provider) and social decentralization (the number of validator nodes per operator). Feel free to explore these metrics using the [Polkawatch App](https://polkadot.polkawatch.app/){target=\_blank}

3. **On-Chain Treasury and Governance** - known as [OpenGov](link to opengov overview page){target=\_blank}, this system allows every decision to be made through public referenda, enabling any token holder to cast a vote
```

### Polkadot's Blockspace

Polkadot 1.0’s design allows for the commoditization of blockspace.

Blockspace is the capacity of a blockchain to finalize and commit operations, encompassing its security, computing, and storage capabilities. Its characteristics can vary across different blockchains, affecting **security**, **flexibility**, and **availability**.

> - **Security** measures the robustness of blockspace in Proof-of-Stake (PoS) networks, linked to the stake locked on validator nodes, the variance in stake among validators, and the total number of validators. It also considers social centralization (how many validators are owned by single operators) and physical centralization (how many validators run on the same service provider).
> 
> - **Flexibility** reflects the functionalities and types of data that can be stored, with high-quality data essential to avoid bottlenecks in critical processes.
> 
> - **Availability** indicates how easily users can access blockspace. It should be easily accessible, allowing diverse business models to thrive, ideally regulated by a marketplace based on demand and supplemented by options for "second-hand" blockspace.

Reviewer: ```suggestion
- **Security** - measures the robustness of blockspace in Proof-of-Stake (PoS) networks, linked to the stake locked on validator nodes, the variance in stake among validators, and the total number of validators. It also considers social centralization (how many validators are owned by single operators) and physical centralization (how many validators run on the same service provider)

- **Flexibility** - reflects the functionalities and types of data that can be stored, with high-quality data essential to avoid bottlenecks in critical processes

- **Availability** - indicates how easily users can access blockspace. It should be easily accessible, allowing diverse business models to thrive, ideally regulated by a marketplace based on demand and supplemented by options for "second-hand" blockspace
```

Polkadot is built on core blockspace principles, but there’s room for improvement. Currently, tasks like balance transfers, staking, and governance are managed on the relay chain. 

Delegating these responsibilities to [system chains](add link to System Chains section) could enhance flexibility and allow the relay chain to concentrate on providing shared security and interoperability.

Currently, blockspace is accessible only through auctions, and the auction winner gains access to a "freighter of blocks," whether needed or not. This setup creates high entry barriers and can lead to wasted energy and resources.

!!!note
    For more information about blockspace see this [interview](https://www.youtube.com/watch?v=e1vISppPwe4){target=\_blank} with **Robert Habermeier** as well as this [article](https://www.rob.tech/blog/polkadot-blockspace-over-blockchains/){target=\_blank} by him.

> ## Protocol Componets

Reviewer: I would entirely remove the content of the Protocol Components section, we can leave the titles of each component and link them to their specific pages. We have pages for accounts, cryptography, runtime upgrades, extrinsics,etc. If some of the content here was not explained in the existing pages it can be added there

Author: yes, perfect. I agree

Reviewer 2: @themacexpert looks like this needs to be removed still

Reviewer 4: Took this stuff out. We do handle all of those things on their own pages in the updated IA

DOT is the native token of the Polkadot network, much like BTC is for Bitcoin and Ether is for the Ethereum blockchain.

### Redomination

Polkadot conducted a community poll, which ended on **27 July 2020** at block **888,888** to decide whether to redenominate the DOT token. 

The stakeholders chose to redenominate the token, changing the value of 1 DOT from 1e12 plancks to 1e10 plancks. 

Importantly, this did not affect the total number of base units (plancks) in the network, only how a single DOT is represented.

The redenomination became effective 72 hours after transfers were enabled, occurring at block **1,248,328** on 21 August 2020 around 16:50 UTC.

## The Planck Unit

The smallest unit of account balance on Substrate-based blockchains (such as Polkadot and Kusama) is called **Planck**, named after the **Planck Length**, the smallest measurable distance in the physical universe. 

Similar to how BTC's smallest unit is the **Satoshi** and ETH's is the **Wei**, Polkadot’s native token DOT is equal to 1e10 Planck, while Kusama’s native token KSM equals 1e12 Planck.

## What are the uses of DOT?

DOT serves four primary functions within the Polkadot network:

> 1. **Governance**: It is used to participate in the governance of the network.
> 2. **Staking**: DOT is staked to support the network’s operation and security.
> 3. **Bonding**: It is bonded to connect new chains to Polkadot as parachains.
> 4. **Buying Coretime**: used to purchase coretime in-bulk or on-demand and access the  chain to benefit from Polkadot's security and interoperability

Reviewer: ```suggestion
1. **Governance** - it is used to participate in the governance of the network
2. **Staking** - DOT is staked to support the network’s operation and security
3. **Bonding** - it is bonded to connect new chains to Polkadot as parachains
4. **Buying Coretime** - used to purchase coretime in-bulk or on-demand and access the  chain to benefit from Polkadot's security and interoperability
```

Additionally, DOT can serve as a transferable token. For example, DOT held in the Treasury can be allocated to teams developing projects that benefit the Polkadot ecosystem.

> ## FAQ

Reviewer: Im not sure if it makes sense to have the FAQ section

Author: section removed
